### PR TITLE
fix: change log level when err is `http.ErrServerClosed`

### PR DESCRIPTION
**Pull request by wafuwafu13**
[ListenAndServe](https://pkg.go.dev/net/http#Server.ListenAndServe) always returns a non-nil error.
If error is `http.ErrServerClosed`, server is not failed to start. 
So log level sholud be Info not Error if error is `http.ErrServerClosed`.

### CODE REVIEW

**Review by alpeb — COMMENTED** [review]
Nice! Some checks are a little flaky, so triggering them again.

File: controller/cmd/destination/main.go [hunk 1]
```diff
@@ -56,7 +58,11 @@ func Main(args []string) {
 	go func() {
 		log.Infof("starting admin server on %s", *metricsAddr)
 		if err := adminServer.ListenAndServe(); err != nil {
-			log.Errorf("failed to start destination admin server: %s", err)
+			if errors.Is(err, http.ErrServerClosed) {
+				log.Infof("destination admin server status is shutdown or close: %s", err)
```
Comment (mateiidavid) [inline controller/cmd/destination/main.go:62]: Hm so if we're handling `SIGTERM` we'd end up here, right? I wonder if it's just better to say something like "received shutdown signal" and leave it at that. I'll have to run it myself to see exactly what `err` evaluates to here :D

**Comment by mateiidavid** [comment]
@wafuwafu13 do you mind pulling in the latest main into your branch? That should fix the `CodeQL` failure. :)

**Comment by wafuwafu13** [comment]
@mateiidavid 
Thank you for telling me, but I'm traveling so I would like to do that next week.

File: controller/cmd/destination/main.go (hunk 1 above)
Comment (mateiidavid) [inline controller/cmd/destination/main.go:62]: Looking at the docs, the error printed for graceful shutdown (i.e. `http.ErrServerClosed`) is something like:

```
"http: Server closed"
```

[ref](https://pkg.go.dev/net/http#ErrServerClosed)

Can we maybe change this to:

```
log.Infof("shutting down admin server on %s", *metricsAddr)
```

Otherwise our log message will look like this:

```
destination admin server status is shutdown or close: http: Server closed
```

File: controller/api/destination/endpoint_profile_translator_test.go [hunk 2]
```diff
@@ -1,8 +1,10 @@
 package destination
 
 import (
+	"errors"
```
Comment (mateiidavid) [inline controller/api/destination/endpoint_profile_translator_test.go:4]: Hmm, for some reason this got picked up even though it's not part of this change. Do you mind pulling latest main again?

```
git pull <upstream> main
```

should do the trick. If it doesn't, you can perhaps do a soft reset and checkout the file.

File: controller/cmd/destination/main.go [hunk 3]
```diff
@@ -56,7 +58,11 @@ func Main(args []string) {
 	go func() {
 		log.Infof("starting admin server on %s", *metricsAddr)
 		if err := adminServer.ListenAndServe(); err != nil {
-			log.Errorf("failed to start destination admin server: %s", err)
+			if errors.Is(err, http.ErrServerClosed) {
+				log.Infof("shutting down admin server on %s", *metricsAddr)
+			} else {
+				log.Errorf("failed to start destination admin server: %s", err)
```
Comment (olix0r) [inline controller/cmd/destination/main.go:64]:
```suggestion
			if errors.Is(err, http.ErrServerClosed) {
				log.Infof("Admin server closed (%s)", *metricsAddr)
			} else {
				log.Errorf("Admin server error (%s): %w", *metricsAddr, err)
```

Some minor nits: %w is necessary to prevent log injection.

**Review by olix0r — APPROVED** [review]

File: controller/cmd/destination/main.go [hunk 4]
```diff
@@ -56,7 +58,11 @@ func Main(args []string) {
 	go func() {
 		log.Infof("starting admin server on %s", *metricsAddr)
 		if err := adminServer.ListenAndServe(); err != nil {
-			log.Errorf("failed to start destination admin server: %s", err)
+			if errors.Is(err, http.ErrServerClosed) {
+				log.Infof("Admin server closed (%s)", *metricsAddr)
+			} else {
+				log.Errorf("Admin server error (%s): %w", *metricsAddr, err)
```
Comment (olix0r) [inline controller/cmd/destination/main.go:64]:
```suggestion
				log.Errorf("Admin server error (%s): %s", *metricsAddr, err)
```
